<?xml version="1.0"?>
<launch>

  <node name="visualizer" pkg="rviz" type="rviz" args="-d $(find traj_gen)/rviz.rviz"/>

  <!--  YOLO detector-->
  <param name="/inference_rate" value="25"/>
  <param name="/img_in_topic" value="/thermal/image_raw"/>
  <param name="/img_in_width" value="640"/>
  <param name="/img_in_height" value="480"/>
  <param name="/img_out" value="true"/>
  <param name="/img_out_topic" value="/detected"/>
  <param name="/bbox_topic" value="/bboxes"/>
  <param name="/confidence_threshold" value="0.8"/>
  <param name="/nms_threshold" value="0.4"/>
  <param name="/class_file" value="$(find traj_gen)/scripts/class.txt"/>
  <param name="/weight_file" value="$(find traj_gen)/scripts/yolov4-tiny-3l-obj_final.weights"/>
  <param name="/cfg_file" value="$(find traj_gen)/scripts/yolov4-tiny-3l-obj.cfg"/>
  <node name="person_detector" pkg="traj_gen" type="ros_opencv_dnn.py" output="screen"/>

  <!--  MAIN code: traj generator-->
  <param name="/thermal_f_x" value="646.02311"/>
  <param name="/thermal_f_y" value="645.2029"/>
  <param name="/thermal_c_x" value="317.59295"/>
  <param name="/thermal_c_y" value="252.3914"/>
  <param name="/thermal_hfov" value="1.0239"/> 

  <param name="/depth_max_range" value="6.0"/>
  <param name="/tof_f_x" value="569.483"/>
  <param name="/tof_f_y" value="568.684"/>
  <param name="/tof_c_x" value="325.065"/>
  <param name="/tof_c_y" value="231.443"/>
  <param name="/tof_hfov" value="0.9948"/> 

  <param name="/depth_topic" value="/cubeeye/scube/depth_raw"/>
  <param name="/body_base" value="base_link"/>
  <param name="/fixed_frame" value="map"/>

  <param name="/real_width" value="0.5"/> <!-- real person width in meter -->

  <param name="/queuing_time" value="0.7"/>
  <param name="/point_numbers" value="10"/>
  <param name="/traj_gen_hz" value="2.0"/>
  <param name="/traj_gen_hz2" value="3.333"/>
  <param name="/min_dist" value="0.9"/>
  <param name="/max_dist" value="2.5"/>
  <param name="/corridor_r" value="0.9"/>
  <param name="/v_max" value="5.0"/>
  <param name="/a_max" value="10.0"/>
  <param name="/n_order" value="7"/>
  <param name="/min_size" value="2"/>
  <param name="/predict_time" value="1.4"/>
  <param name="/min_prob" value="0.91"/>

  <param name="/k_pos" value="0.4"/>
  <param name="/k_vel" value="1.6"/>
  <param name="/k_ff" value="0.01"/>

  <param name="/debug" value="true"/>
  <param name="/deep_debug" value="false"/>

  <node name="traj_gen_node" pkg="traj_gen" type="traj_gen_node" output="screen"/>
  <!-- launch-prefix="gdb -ex run --> <!--args" -->

</launch>
